CMSgov/mint-app
# [NOREF] Use new graphql-codegen tool

# EASI-000

## Changes and Description

- Introduce new `queriesCodegen` folder, where all queries will reside for the new codegen tool

The high level approach used here is to introduce `graphql-codegen`, which is more supported than the apollo codegen tool we're using (see [this issue](https://github.com/apollographql/apollo-tooling/issues/2053)). The goal is to introduce a way to slowly cut over queries to this new codegen tool without having to completely and immediately deprecate apollo codegen.

Looking for feedback on the following:
- Organization of files (one file per query? Multiple? Both are supported!)
- Naming (Not sure if LOVE `queriesCodegen` - could be convinced to name this something else
- Any additional configs/settings that the new codegen tool might provide that we should enable here!

## How to test this change

- Run `scripts/dev gql` to ensure no files change
- Try adding a new query to `src/queriesCodegen` and using it within the app!

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
